app/client/src/components/app.tsx

```diff
@@ -21,13 +21,29 @@ import Loading from "components/loading";
 import Welcome from "components/welcome";
 import Dashboard from "components/dashboard";
 import ConfirmationDialog from "components/confirmationDialog";
+import Helpdesk from "routes/helpdesk";
 import AllRebateForms from "routes/allRebateForms";
 import NewRebateForm from "routes/newRebateForm";
 import ExistingRebateForm from "routes/existingRebateForm";
 import NotFound from "routes/notFound";
 import { useUserState, useUserDispatch } from "contexts/user";
 import { useDialogDispatch, useDialogState } from "contexts/dialog";
 
+type HelpdeskAccess = "idle" | "pending" | "success" | "failure";
+
+export function useHelpdeskAccess() {
+  const [helpdeskAccess, setHelpdeskAccess] = useState<HelpdeskAccess>("idle");
+
+  useEffect(() => {
+    setHelpdeskAccess("pending");
+    fetchData(`${serverUrl}/api/helpdesk-access`)
+      .then((res) => setHelpdeskAccess("success"))
+      .catch((err) => setHelpdeskAccess("failure"));
+  }, []);
+
+  return helpdeskAccess;
+}
```

Comment:
This is where the verification request/response occurs the client to the server. I setup this as a custom hook as it's called in 1) the `<Dashboard />` component to determine if the button/link to the `/dashboard` route should be displayed, and 2) on the `/dashboard` page itself (in case a user visits there directly). On the `/dashboard` route itself, nothing is shown to the user until the API call confirms the user has access. If it responds that they don't they're redirected back the `/` route.

@dpgalloway nothing for you to do here – just wanted to point out how I set it up.